test/Mono.Linker.Tests.Cases/DataFlow/XmlAnnotations.cs

```diff
@@ -13,7 +13,6 @@ namespace Mono.Linker.Tests.Cases.DataFlow
 {
 	[SkipKeptItemsValidation]
 	[SetupLinkAttributesFile ("XmlAnnotations.xml")]
-	[ExpectedWarning ("IL2031", "System.DoesNotExistAttribute", FileName = "XmlAnnotations.xml")]
 	[LogDoesNotContain ("IL2067: Mono.Linker.Tests.Cases.DataFlow.XmlAnnotations.ReadFromInstanceField():*", true)]
 	class XmlAnnotations
```

Comment:
We should add a test case similar to the one in #1915, DynamicDependency not only makes use of the overload feature that this PR fixes, but also interacts with the caches in the linker so it triggers an action that can be verified by the ResultChecker

Reply:
Added extra test for DynamicDependency with Type argument into relevant tests